velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -158,7 +156,25 @@ const ::substrait::Expression& VeloxToSubstraitExprConvertor::toSubstraitExpr(
         typeConvertor_->toSubstraitType(arena, callTypeExpr->type()));
 
   } else {
-    VELOX_NYI("Unsupported function name '{}'", functionName);
+    // For today's version of Substrait, you'd have to use IfThen if you need
+    // the switch cases to be call typed expression.
+
+    size_t inputsSize = callTypeExpr->inputs().size();
+    bool hasElseInput = inputsSize % 2 == 1;
+
+    auto ifThenExpr = substraitExpr->mutable_if_then();
+    size_t last = hasElseInput ? inputsSize - 1 : inputsSize;
```

Comment:
This variable is not needed. You can use inputsSize instead.

Reply:
updated